app/models/provider.rb

```diff
@@ -36,7 +38,11 @@ class Provider < ApplicationRecord
   has_many :sites
   has_many :enrichments, foreign_key: :provider_code, primary_key: :provider_code, class_name: "ProviderEnrichment"
   def address_info
-    (enrichments.with_address_info.last || self)
+    @last_enrichment = enrichments.with_address_info.last
+    @address_info = (@last_enrichment || self)
       .slice('address1', 'address2', 'address3', 'address4', 'postcode')
+    @region_code = @last_enrichment.present? ? @last_enrichment[:region_code] : self.region_code_before_type_cast
```

Comment:
I'd suggest you actually include the `region_code` enum into `ProviderEnrichment` instead of doing this juggling around here. You may be able to then do:

```
(enrichments.with_address_info.last || self)
  .attributes_before_type_cast
  .slice('address1', 'address2', 'address3', 'address4', 'postcode', 'region_code')
```